src/core/qgsbrowsermodel.cpp

```diff
@@ -123,42 +123,9 @@ void QgsBrowserModel::addRootItems()
   mRootItems << vols;
 #endif
 
-  // container for displaying providers as sorted groups (by QgsDataProvider::DataCapability enum)
-  QMap<int, QgsDataItem *> providerMap;
-
   Q_FOREACH ( QgsDataItemProvider *pr, QgsApplication::dataItemProviderRegistry()->providers() )
   {
-    int capabilities = pr->capabilities();
-    if ( capabilities == QgsDataProvider::NoDataCapabilities )
-    {
-      QgsDebugMsgLevel( pr->name() + " does not have any dataCapabilities", 4 );
-      continue;
-    }
-
-    QgsDataItem *item = pr->createDataItem( QString(), nullptr );  // empty path -> top level
-    if ( item )
-    {
-      // Forward the signal from the root items to the model (and then to the app)
-      connect( item, &QgsDataItem::connectionsChanged, this, &QgsBrowserModel::connectionsChanged );
-      QgsDebugMsgLevel( "Add new top level item : " + item->name(), 4 );
-      setupItemConnections( item );
-      providerMap.insertMulti( capabilities, item );
-    }
-  }
-
-  // add as sorted groups by QgsDataProvider::DataCapability enum
-  Q_FOREACH ( int key, providerMap.uniqueKeys() )
-  {
-    QList<QgsDataItem *> providerGroup = providerMap.values( key );
-    if ( providerGroup.size() > 1 )
-    {
-      std::sort( providerGroup.begin(), providerGroup.end(), cmpByDataItemName_ );
-    }
```

Comment:
It seems the sorting got dropped - is that intentional?

Reply:
I believe it's redundant now - sorting is done by the QgsBrowserProxyModel by item sort key.